clients/geth/specular/rollup/services/sequencer/sequencer.go

```diff
@@ -22,14 +24,21 @@ import (
 
 const timeInterval = 10 * time.Second
 
+func funcName() string {
+    pc, _, _, _ := runtime.Caller(1)
+	nameFull := runtime.FuncForPC(pc).Name()    
+    splitInput := strings.Split(nameFull, "/")
+    return splitInput[len(splitInput)-1]
+}
+
 func RegisterService(stack *node.Node, eth services.Backend, proofBackend proof.Backend, cfg *services.Config, auth *bind.TransactOpts) {
 	sequencer, err := New(eth, proofBackend, cfg, auth)
 	if err != nil {
-		log.Crit("[Sequencer: RegisterService] Failed to register the Rollup service", "err", err)
+		log.Crit("["+funcName()+"] Failed to register the Rollup service", "err", err)
```

Comment:
ok i didn't realize this would look so ugly

can we abstract this away—should we just define our own `log` with a function for each level that calls into the corresponding geth functions, passing `funcName()` in? note: would need to do `runtime.Caller(2)` in `funcName()` in that case i believe

Reply:
Logs util added and logs updated as per geth logs